src/Path.Drag.js

```diff
@@ -171,18 +183,31 @@ L.Handler.PathDrag = L.Handler.extend( /** @lends  L.Path.Drag.prototype */ {
     var diff = transformation.untransform(px, scale)
       .subtract(transformation.untransform(L.point(0, 0), scale));
 
-    // console.time('transform');
+    return projection.unproject(projection.project(point)._add(diff));
+  },
+
+  /**
+   * Applies transformation, does it in one sweep for performance,
+   * so don't be surprised about the code repetition.
+   *
+   * [ x ]   [ a  b  tx ] [ x ]   [ a * x + b * y + tx ]
+   * [ y ] = [ c  d  ty ] [ y ] = [ c * x + d * y + ty ]
+   *
+   * @param {Array.<Number>} matrix
+   */
+  _transformPoints: function(matrix) {
+    var path = this._path;
+    var i, len;
+
+    var px = L.point(matrix[4], matrix[5]);
 
     // all shifts are in-place
     if (path._point) { // L.Circle
-      path._latlng = projection.unproject(
-        projection.project(path._latlng)._add(diff));
+      path._latlng = this._transformPoint(matrix, path._latlng);
       path._point._add(px);
     } else if (path._originalPoints) { // everything else
       for (i = 0, len = path._originalPoints.length; i < len; i++) {
-        latlng = path._latlngs[i];
-        path._latlngs[i] = projection
-          .unproject(projection.project(latlng)._add(diff));
+        path._latlngs[i] = this._transformPoint(matrix, path._latlngs[i]);
```

Comment:
Could you please provide me, how did you measure this?

I've just made a small benchmark and didn't notice difference. Actually, your original version is 0.005 longer in average in my tests ;)

######  my PR version

time |
------------
0.095ms |
0.098ms |
0.066ms |
0.054ms |
0.046ms |
0.055ms |
0.046ms |
0.055ms |
0.062ms |
0.055ms |
0.053ms |
0.055ms |
0.054ms |
0.053ms |
0.081ms |
0.051ms |
0.057ms |

`0.06094117647ms average`

######  original version

time |
--------
0.103ms |
0.084ms |
0.068ms |
0.057ms |
0.058ms |
0.056ms |
0.058ms |
0.049ms |
0.058ms |
0.059ms |
0.065ms |
0.063ms |
0.073ms |
0.057ms |
0.048ms |
0.058ms |
0.059ms |
0.088ms |

`0.0645ms average`